urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/utils/IntervalsSetMerger.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package net.personaltt.utils;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+/**
+ * Intervals set merger provides method and interface for merging two sets of 
+ * intervals represented by intervals stops lists
+ * @author docx
+ * @param  <T> Type of intervals stops
+ * @param  <V> Type of interval values
+ */
+public class IntervalsSetMerger<T extends Comparable,V1,V2> {
+    
+    Iterator<Map.Entry<T, V1>> this_iterator;
+    Iterator<Map.Entry<T, V2>> second_iterator;
+
+    public IntervalsSetMerger(Iterator<Entry<T, V1>> this_iterator, Iterator<Entry<T, V2>> second_iterator) {
+        this.this_iterator = this_iterator;
+        this.second_iterator = second_iterator;
+    }
+    
+    /*
+     * Merge walk througt edges from both intervals sets (this and given) and calls merge function 
+     * implementation for each of them
+     */
+    public <R> TreeMap<T, R> merge(MergeFunction<R, V1, V2> operation) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.